### PR TITLE
Fix: Negative blood levels now kill you

### LINKED ISSUE #38024: Humans don't die with negative blood
id 88397

ok so pro strats here
if you have the blood deficiency thing and you get changed to a race that doesn't get effected by bloodloss you can go into negative blood and even when you switch back to human you don't die (given you are below 0% at that point)

### PR DESCRIPTION
**Pull request by CitrusGender**
This should stop the blood loss with the bloodloss trait and kills you if you go below 0% blood. (or atleast, nobody should ever be able to feasibly get to -9999% blood.) I wasn't able to find any of the toxic damage stuff/blood loss and the only thing that caused the negative blood loss in the first place was the trait.

:cl: 
fix: Kills you if you're under 0% blood and above -9999% blood level (shouldn't be possible through normal means to be below that.)
fix: Stops bloodloss trait from happening if your race contains the NO_BLOOD
/:cl:

Fixes #38024

### CODE REVIEW

File: code/modules/mob/living/blood.dm [hunk 1]
```diff
@@ -68,7 +68,7 @@
 				if(prob(15))
 					Unconscious(rand(20,60))
 					to_chat(src, "<span class='warning'>You feel extremely [word].</span>")
-			if(0 to BLOOD_VOLUME_SURVIVE)
+			if(-9999 to BLOOD_VOLUME_SURVIVE)
```
Comment (SpaceManiac) [inline code/modules/mob/living/blood.dm]: `-INFINITY`

File: code/datums/traits/negative.dm [hunk 2]
```diff
@@ -9,7 +9,10 @@
 	medical_record_text = "Patient requires regular treatment for blood loss due to low production of blood."
 
 /datum/quirk/blooddeficiency/on_process()
-	quirk_holder.blood_volume -= 0.275
+	var/mob/living/carbon/human/H = quirk_holder
+	if(NOBLOOD in H.dna.species.species_traits) //can't lose blood if your species doesn't have any
+		return
+	else quirk_holder.blood_volume -= 0.275
```
Comment (SpaceManiac) [inline code/datums/traits/negative.dm]: Indented block, not same line

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Comment by CitrusGender** [comment]
alright, hopefully that's good. I think I resolved the merge conflict too

**Review by SpaceManiac — APPROVED** [review]

**Comment by Arianya** [comment]
Worth double checking to make sure nothing like vampire relies on negative blood levels

**Comment by KorPhaeron** [comment]
Why are people reaching negative blood levels at all

**Comment by SpaceManiac** [comment]
Automatic bloodloss quirk combined with a species that doesn't use blood (e.g. golem).

**Comment by Spaceinaba** [comment]
@KorPhaeron don't question my ability to find the most niche bugs ever

**Pull request merged**